qiskit/pulse/schedule.py

```diff
@@ -186,6 +188,87 @@ def flatten(self) -> 'ScheduleComponent':
         """Return a new schedule which is the flattened schedule contained all `instructions`."""
         return ops.flatten(self)
 
+    def filter(self, *filter_funcs: List[Callable],
+               channels: Iterable[int] = [],
+               instruction_types: Iterable[abc.ABCMeta] = [],
```

Comment:
What do you think about adding an `instructions` and `commands` default filter that matches specific instances of an `Instruction/Command`? For example, it might be useful in testing to find out what times a given command happens at.

Reply:
That feels like a slightly different, easier task -- this method might be a bit heavy handed for it. By that I mean, if you wanted to do that, you wouldn't think to write a filter and pass your schedule through it, and then loop through the results to get the times, you'd just write a very simple for loop and collect all times as you go through. The straight python way seems easier, for the writer and reader, and is probably faster.

Perhaps I need more practical examples of the use case? I'm happy to add a separate `get_times` search method to do this (since time/existence would seem to be the only new info you could get once you have an exact instruction instance).